controllers/topics/broadcastMessage.js

```diff
@@ -0,0 +1,117 @@
+/* 
+    Send message to all users who join a topic
+*/
+const momentTz = require('moment-timezone');
+const {v4: uuidv4} = require('uuid');
+const UsersFunction = require('../../databases/functions/users');
+const {followTopicQueue} = require('../../services/redis/queue');
+const {User, CommunityMessageFormat, UserTopic} = require('../../databases/models');
+const {Op} = require('sequelize');
+
+const sendMessageToQueue = async (user_id, topic_id, community_message_format_id, delay) => {
+  const data = {
+    user_id,
+    topic_id,
+    community_message_format_id
+  };
+  let currentTime = momentTz().tz('America/Los_Angeles');
+  let requiredTime = momentTz().tz('America/Los_Angeles').add(delay, 'minutes');
+
+  const diffTime = requiredTime.diff(currentTime, 'milliseconds');
+
+  const options = {
+    jobId: uuidv4(),
+    removeOnComplete: true,
+    delay: diffTime
+  };
+  try {
+    await followTopicQueue.add(data, options);
+  } catch (e) {
+    console.log('error', e);
+  }
+};
+
+const calculateDelay = (joinDate, delay) => {
+  let currentTime = momentTz().tz('America/Los_Angeles');
+  let requiredTime = momentTz(joinDate).tz('America/Los_Angeles');
+
+  let diffTime = requiredTime.diff(currentTime, 'days');
+  diffTime = Math.abs(diffTime + delay);
+  return diffTime == 0 ? delay * 60 * 1000 : diffTime * 24 * 60 * 60 * 1000;
+};
+
+module.exports = async (req, res) => {
+  const {userId} = req;
+  const {communityMessageFormatId, topicId, targetUserId, targetDate, delay} = req.body;
+
+  //   check if the the user is anonymous
+  const detailTokenUser = await UsersFunction.findUserById(User, userId);
+  if (!detailTokenUser) {
+    return res.status(404).json({message: 'User not found'});
+  }
+  if (detailTokenUser.is_anonymous) {
+    return res.status(403).json({message: 'Anonymous user cannot send message'});
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling is missing for the case when `UsersFunction.findUserById(User, userId)` fails due to reasons other than the user not being found (e.g., database connection issues). Consider adding a try-catch block around this call to handle potential errors.

```diff
-  const detailTokenUser = await UsersFunction.findUserById(User, userId);
+  let detailTokenUser;
+  try {
+    detailTokenUser = await UsersFunction.findUserById(User, userId);
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({message: 'Internal server error'});
+  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->